kpops/component_handlers/kafka_connect/model.py

```diff
@@ -1,30 +1,47 @@
 from enum import Enum
 from typing import Any, Literal
 
-from pydantic import BaseConfig, BaseModel, Extra
+from pydantic import BaseConfig, BaseModel, Extra, Field, validator
 from typing_extensions import override
 
-from kpops.utils.docstring import describe_object
-from kpops.utils.pydantic import CamelCaseConfig, DescConfig
+from kpops.utils.pydantic import CamelCaseConfig, DescConfig, to_dot
 
 
 class KafkaConnectorType(str, Enum):
     SINK = "sink"
     SOURCE = "source"
 
 
-class KafkaConnectConfig(BaseModel):
+class KafkaConnectorConfig(BaseModel):
     """Settings specific to Kafka Connectors"""
 
+    connector_class: str
+    name: str = Field(default=..., hidden_from_schema=True)
+
     class Config(DescConfig):
         extra = Extra.allow
+        alias_generator = to_dot
 
         @override
-        @staticmethod
-        def schema_extra(schema: dict[str, Any], model: type[BaseModel]) -> None:
-            schema["description"] = describe_object(model.__doc__)
+        @classmethod
+        def schema_extra(cls, schema: dict[str, Any], model: type[BaseModel]) -> None:
+            super().schema_extra(schema, model)
```

Comment:
Isn't it weird to use super() for a class method ? super is for instances. Shouldn't be `cls.schema_extra`?

Reply:
Wouldn't that be a recursive call?

Reply:
hm, I am not aware of any implications using `super()`. essentially it would be something like `super(cls, cls)` but we can omit the arguments

Reply:
@irux 

If I understood your suggestion correctly, `cls.schema_extra` should work equivalently to the example bellow:

```
>>> class A:
...     @classmethod
...     def print_class(cls):
...             print("Class method of A")
...
>>> A.print_class()
Class method of A
>>> class B(A):
...     @classmethod
...     def print_class(cls):
...             print("Class method of B")
...             cls.print_class()
...
>>> B.print_class
<bound method B.print_class of <class '__main__.B'>>
>>> B.print_class()
Class method of B
Class method of B
Class method of B
Class method of B
Class method of B
Class method of B
```